SilverSolver/assignment5/unknown_alphabet.py

```diff
@@ -0,0 +1,49 @@
+from collections import defaultdict
+
+def dfs(ordered_vertices, edges, color, v):
+    if color[v] == 1:
+        return (ordered_vertices, None)
+    if color[v] == 2:
+        return (ordered_vertices, color)
+    color[v] = 1
+    if v in edges.keys():
+        for v_ in edges[v]:
+            ordered_vertices, color = dfs(ordered_vertices, edges, color, v_)
+            if v_ in edges.keys() and color == None:
+                return (ordered_vertices, None)
+    ordered_vertices.append(v)
+    color[v] = 2
+    return (ordered_vertices, color)
+
+def topological_sort(edges, symbols):
+    ordered_vertices = []
+    N = len(symbols)
+    color = dict()
+    for v in symbols:
+        color[v] = 0
```

Comment:
stuff like this could look cleaner as
color = dict.fromkeys(symbols, 0)

Reply:
Yes, I agree